src/routes/run.rs

```diff
@@ -130,75 +192,21 @@ async fn find_for_test(
     id: web::Path<String>,
     web::Query(query): web::Query<RunQueryIncomplete>,
     pool: web::Data<db::DbPool>,
-) -> impl Responder {
+) -> HttpResponse {
```

Comment:
I thought about doing that but figured that, regardless of whether things go right or wrong, I'm returning an HttpResponse (either with the requested data or an error message) so it didn't make sense to do something like `Result<HttpResponse, HttpResponse>`

Reply:
is there a fundamental rust problem with Result<HttpResponse, HttpResponse>? how does rust determine if the return was error or not under the hood?

Reply:
It would be totally find to do that (in fact, I think there are a couple places in the code where I am doing that and will probably change), but the purpose of returning Result from a function is to force the calling function to handle the error (or continue to propagate it up the call stack).  For these particular functions, they are being called directly by actix (the library I'm using for http stuff) to handle requests, so whatever is returned by them is what is going to be returned to the user.  So, I'm actually handling the errors within these functions by logging them and returning an appropriate HttpResponse.  If I were to return a Result<HttpResponse, HttpResponse>, it wouldn't change the functionality in any way because actix will just deconstruct the Result into its inner type and return that to the user.  So, I figured it was cleaner to just say, "This thing returns an HttpResponse."